### PR TITLE
Making dotnet.exe a .Net Core app so that it can compile itself.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Changing some API uses to only the ones available via NuGet and adding
the extra packages, references, and dependencies so that dotnet can
compile itself.

### CODE REVIEW

**Comment by ahsonkhan** [comment]
About issue #242 /issue #218

File: demos/dotnet/dotnet/dotnet.cs [hunk 1]
```diff
@@ -712,13 +735,6 @@ public static bool RestorePackagesAction(ProjectProperties properties, Log log)
             Console.WriteLine(e);
             return false;
         }
-        finally
-        {
-            if (process != null)
-            {
-                process.Close();
```
Comment (weshaggard) [inline demos/dotnet/dotnet/dotnet.cs]: You should switch this Close call to a Dispose call.

Comment (ahsonkhan) [inline demos/dotnet/dotnet/dotnet.cs]: I removed the Close call already. Is disposal necessary?

File: demos/dotnet/dotnet/dotnet.cs [hunk 2]
```diff
@@ -651,7 +671,10 @@ static class NugetAction
 {
     static void CreateNugetConfig(ProjectProperties properties)
     {
-        using (var file = new StreamWriter(Path.Combine(properties.ToolsDirectory, "nuget.config"), false))
+        string fileName = Path.Combine(properties.ToolsDirectory, "nuget.config");
+        FileStream fs = null;
+        fs = new FileStream(fileName, FileMode.Create);
```
Comment (KrzysztofCwalina) [inline demos/dotnet/dotnet/dotnet.cs]: Why the above are two separate lines of code?

**Comment by KrzysztofCwalina** [comment]
looks good

File: demos/dotnet/dotnet/dotnet.cs (hunk 1 above)
Comment (weshaggard) [inline demos/dotnet/dotnet/dotnet.cs]: Yes it is a good idea to call Dispose. You should just add a using statement around the Process usage.

File: demos/dotnet/dotnet/dotnet.cs (hunk 2 above)
Comment (ahsonkhan) [inline demos/dotnet/dotnet/dotnet.cs]: Agreed. Merged into single line.

File: demos/dotnet/dotnet/dotnet.cs (hunk 1 above)
Comment (ahsonkhan) [inline demos/dotnet/dotnet/dotnet.cs]: Good idea. Done.

**Pull request merged**